### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by manueltuero**
## Summary
* Add a `/users` POST endpoint

## Trello Card
https://trello.com/c/HBBYGAQG

### CODE REVIEW

File: app/models/user.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        field: 'last_name',
+        type: DataTypes.STRING
+      },
+      email: {
+        allowNull: false,
+        type: DataTypes.STRING,
+        unique: true,
+        validate: {
+          isEmail: true,
+          contains: '@wolox.com.ar'
```
Comment (jersonlopez) [inline app/models/user.js]: what happend if my email is: wolox.co/cl/com?

File: app/services/users.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const { users } = require('../models'),
+  logger = require('../logger'),
+  errors = require('../errors'),
+  bcrypt = require('bcryptjs');
+const saltRounds = 10;
```
Comment (jersonlopez) [inline app/services/users.js]: this is a magic number, it would be better as an environment variable

File: app/services/users.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    let user = await users.findOne({
+      where: {
+        email: data.email
+      }
+    });
+    let message = '';
+    if (user) {
+      message = `The user "${user.email}" already exists`;
+      logger.error(message);
+      return errors.userAlreadyExistsError(message);
```
Comment (jersonlopez) [inline app/services/users.js]: return is not necessary, work with `throw`

File: app/controllers/users.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const { createUser } = require('../services/users');
+
+exports.createUser = async (req, res, next) => {
+  try {
+    const userName = await createUser(req.query);
+    res.status(200).send(userName);
```
Comment (jersonlopez) [inline app/controllers/users.js]: what does mean status 200, is the corrects status for this action?

File: app/models/user.js (hunk 1 above)
Comment (manueltuero) [reply app/models/user.js]: haha it's true, now I change it

File: app/services/users.js (hunk 3 above)
Comment (manueltuero) [reply app/services/users.js]: I fix this in the next PR

**Pull request merged**